galaxy_ng/app/access_control/access_policy.py

```diff
@@ -62,13 +63,34 @@ class CollectionRemoteAccessPolicy(AccessPolicyBase):
 class UserAccessPolicy(AccessPolicyBase):
     NAME = 'UserViewSet'
 
-
-class MyUserAccessPolicy(AccessPolicyBase):
-    NAME = 'MyUserViewSet'
-
     def is_current_user(self, request, view, action):
         return request.user == view.get_object()
 
+    def has_groups_param_obj_perms(self, request, view, action, permission):
+        """
+        Checks if the current user has object-level permission on the ``groups`` object.
+
+        The object in this case is the one specified by the ``remote`` parameter. For example when
+        syncing the ``remote`` parameter is passed in as an argument.
+
+        """
+
+        serializer = UserSerializer(data=request.data, context={"request": request})
+
+        try:
+            serializer.is_valid(raise_exception=True)
+        except Exception as exc:
+            log.exception(exc)
+            raise
+
+        groups = serializer.validated_data.get("groups")
+
+        return request.user.has_perm(permission, groups)
```

Comment:
I don't think this will work. This is is just checking to see if the any of the groups that are being added to the user have the `galaxy.change_user` permission, which doesn't seem right to me.